### PR TITLE
Pokemon

### PR DESCRIPTION
**Pull request by prae014**

### CODE REVIEW

File: graph/generated.go
Comment (Nathavee-mh) [inline graph/generated.go:1]: consider add this file in git ignore

File: graph/model/models_gen.go
Comment (Gromitmugs) [inline graph/model/models_gen.go:1]: generated files are often not pushed onto git
what you should do is to create a `.gitignore` file at the root of the directory
and put on the following lines
```
**_gen.go
**generated.go
```

File: graph/schema.resolvers.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/prae014/pokemon/graph/model"
+)
+
+//TODO: look at ID and stuff
+//TODO: haven't tested update func and whatever came after that
```
Comment (Nathavee-mh) [inline graph/schema.resolvers.go:15]: remove this or fix it

File: server.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	//dsn := "host=localhost user=postgres password=admin dbname=test port=5432 sslmode=disable TimeZone=Asia/Shanghai"
+	dsn := "postgresql://postgres:admin@pokedex_postgres:5432/test"
+	db, err := gorm.Open(postgres.Open(dsn), &gorm.Config{})
+	//db, err := gorm.Open(sqlite.Open("test.db"), &gorm.Config{})
+	if err != nil {
+		panic(err)
+	}
+
+	db.AutoMigrate(&model.Pokemon{}, &model.PokemonType{}, &model.PokemonAbility{})
+	poke1 := model.Pokemon{
```
Comment (Gromitmugs) [inline server.go:35]: HLw-OE

File: graph/schema.resolvers.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+func TypeInputMap(pkmn_type_input []*model.PokemonTypeInput) (pkmn_type []*model.PokemonType) {
```
Comment (Nathavee-mh) [inline graph/schema.resolvers.go:17]: function name should be verb form

File: server.go (hunk 2 above)
Comment (Gromitmugs) [reply server.go:35]: ikusoooo

File: graph/schema.resolvers.go (hunk 3 above)
Comment (Nathavee-mh) [inline graph/schema.resolvers.go:17]: variable name should be camal case like the other variable (pkmn_type_input) 
 add avoid using abbreviation

File: graph/schema.resolvers.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+		Abilities:   AbilityInputMap(input.Abilities),
+	}
+
+	//add to database
+	err := r.DB.Create(&new_pkmn).Error
+	if err != nil {
+		return nil, err
+	}
+	fmt.Println("hereeeee")
+	fmt.Println(new_pkmn.Name)
```
Comment (Nathavee-mh) [inline graph/schema.resolvers.go:60]: remove unused code

File: graph/schema.resolvers.go [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+		return false, nil //FIXME: nil here is a placeholder
+	}
+	return true, nil
+	//panic(fmt.Errorf("not implemented: DeletePokemon - deletePokemon"))
+}
+
+// Pokemons is the resolver for the pokemons field.
+func (r *queryResolver) Pokemons(ctx context.Context) ([]*model.Pokemon, error) {
+	var all_poke []*model.Pokemon
+	r.DB.Preload("Type").Preload("Abilities").Find(&all_poke)
```
Comment (Gromitmugs) [inline graph/schema.resolvers.go:103]: dont forget to handle an error

File: graph/schema.resolvers.go [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+
+	updated_poke := model.Pokemon{
+		ID:          id,
+		Name:        input.Name,
+		Description: input.Description,
+		Category:    input.Category,
+		Type:        TypeInputMap(input.Type),
+		Abilities:   AbilityInputMap(input.Abilities),
+	}
+	r.DB.Save(&updated_poke)
```
Comment (Gromitmugs) [inline graph/schema.resolvers.go:83]: if there is any error variable returned from this function, please check it and handle the error

File: graph/schema.resolvers.go [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+	//panic(fmt.Errorf("not implemented: UpdatePokemon - updatePokemon"))
+}
+
+// DeletePokemon is the resolver for the deletePokemon field.
+func (r *mutationResolver) DeletePokemon(ctx context.Context, id int) (bool, error) {
+	p := r.DB.Delete(&model.Pokemon{}, id)
+	if p.RowsAffected < 1 {
+		fmt.Printf("numbers of row affected: %v\n", p.RowsAffected)
+		fmt.Println("The provided id doesn't exist")
+		return false, nil //FIXME: nil here is a placeholder
```
Comment (Nathavee-mh) [inline graph/schema.resolvers.go:94]: should return an error

**Review by Nathavee-mh — CHANGES_REQUESTED** [review]

**Review by Gromitmugs — CHANGES_REQUESTED** [review]
please read my comments

**Pull request merged**